### PR TITLE
Add small struct DeckKeyword::Location to simplify error reporting

### PR DESCRIPTION
**Pull request by joakim-hove**
Created a smal struct to keep track of the position of a keyword in the input deck; file and linenumber. For keywords which are created by parsing a string the filename will be `<memory string>`.

The location is passed as constructor argument to `DeckKeyword`so that the `DeckKeyword::setLocation()` could be removed - we all love immutable Riiigght?!

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by bska** [comment]
Yuhuu!  This is a very welcome addition.

File: opm/parser/eclipse/Deck/DeckKeyword.hpp [hunk 1]
```diff
@@ -34,6 +34,20 @@ namespace Opm {
 
     class DeckKeyword {
     public:
+
+        class Location {
+        public:
+            std::string filename;
+            std::size_t lineno;
+
+            Location() = default;
```
[the remark this answers is not in the document]
Comment (joakim-hove) [reply opm/parser/eclipse/Deck/DeckKeyword.hpp]: I know I know I know - it is WIP guys ... :smile:

[the remark this answers is not in the document]
Comment (atgeirr) [reply opm/parser/eclipse/Deck/DeckKeyword.hpp]: Didn't notice, sorry. But don't fault me for liking the little class :-)

**Comment by joakim-hove** [comment]
Ready for review

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Looks fine, approved. Merge when green.

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2075 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2075 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2075 please

**Pull request merged**